File: akka-learning-scala/src/main/java/org/spiral/scala/sql/actor/AkkademyDb.scala
package org.spiral.scala.sql.actor

import akka.actor.Actor
import akka.event.Logging
import org.spiral.scala.sql.model.SetRequest

import scala.collection.mutable.HashMap

/**
 * 消息处理
 *
 * @author dengguoqing
 * @date 2019/12/12
 * @since 1.0 Version
 * @copyright spiral
 */
class AkkademyDb extends Actor {
  val map = new HashMap[String, AnyRef]
  val log = Logging(context.system, this)

  override def receive: Receive = {
    case SetRequest(key, value) => {
      log.info("receive SetRequest - key:{} value:{}", key, value)
      map.put(key, value)
    }
    case o => log.info("received unknown message: {}", o)
  }
}

File: akka-learning-java/src/main/java/org/spiral/java/sql/actor/AkkademyDb.java
package org.spiral.java.sql.actor;

import akka.actor.AbstractActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;
import akka.japi.pf.ReceiveBuilder;
import org.spiral.java.sql.model.SetRequest;

import java.util.HashMap;
import java.util.Map;

/**
 * 消息接收响应处理
 *
 * @author dengguoqing
 * @date 2019/12/12
 * @copyright spiral
 * @since 1.0 Version
 */
public class AkkademyDb extends AbstractActor {

    protected final LoggingAdapter log = Logging.getLogger(context().system(), this);

    protected final Map<String, Object> map = new HashMap<>();

    @Override
    public Receive createReceive() {
        return ReceiveBuilder.create().match(SetRequest.class,
                                             message -> {
                                                 log.info("Receive set request", message);
                                                 map.put(message.getKey(),
                                                         message.getValue());
                                             }).matchAny(
                o -> log.info("received unknown message: {}", o)).build();
    }

}
